backend/layers/thirdparty/s3_provider.py

```diff
@@ -48,6 +48,18 @@ def upload_file(self, src_file: str, bucket_name: str, dst_file: str, extra_args
             ExtraArgs=extra_args,
         )
 
+    def delete_files(self, bucket_name: str, object_keys: List[str]):
+        """
+        Deletes the objects `object_keys` from bucket `bucket_name`
+        """
+
+        def chunk_list(items: List[str], chunk_size: int = 1000):  # s3 client delete_objects accepts up to 1000 keys
+            for i in range(0, len(items), chunk_size):
+                yield items[i : i + chunk_size]
+
+        for keys_batch in chunk_list(object_keys):
```

Comment:
We should make the 1000 into a constant.

```suggestion
        AWS_S3_BULK_DELETE_LIMIT = 1000
        for i in range(0, len(object_keys), AWS_S3_BULK_DELETE_MAX):
            keys_batch = object_keys[ i: i+AWS_S3_BULK_DELETE_MAX]
```